lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/S3StorageClient.java

```diff
@@ -34,137 +31,147 @@ public class S3StorageClient extends BaseStorageClient {
   }
 
-  private Document s3ObjectToDoc(S3Object obj) {
-    String fullPath = getFullPath(obj);
+  private Document s3ObjectToDoc(S3Object obj, TraversalParams params) {
+    String fullPath = getFullPath(obj, params);
     String docId = DigestUtils.md5Hex(fullPath);
-    Document doc = Document.create(docIdPrefix + docId);
+    Document doc = Document.create(params.getDocIdPrefix() + docId);
     doc.setField(FileConnector.FILE_PATH, fullPath);
     doc.setField(FileConnector.MODIFIED, obj.lastModified());
     // s3 doesn't have object creation date
     doc.setField(FileConnector.SIZE, obj.size());
 
-    if (getFileContent) {
+    if (params.shouldGetFileContent()) {
       byte[] content = s3.getObjectAsBytes(
-          GetObjectRequest.builder().bucket(bucketOrContainerName).key(obj.key()).build()
+          GetObjectRequest.builder().bucket(getBucketOrContainerName(params)).key(obj.key()).build()
       ).asByteArray();
       doc.setField(FileConnector.CONTENT, content);
     }
 
     return doc;
   }
 
-  private Document s3ObjectToDoc(S3Object obj, InputStream is, String decompressedFullPathStr)
+  private Document s3ObjectToDoc(S3Object obj, InputStream is, String decompressedFullPathStr, TraversalParams params)
       throws IOException {
     String docId = DigestUtils.md5Hex(decompressedFullPathStr);
-    Document doc = Document.create(docIdPrefix + docId);
+    Document doc = Document.create(params.getDocIdPrefix() + docId);
     doc.setField(FileConnector.FILE_PATH, decompressedFullPathStr);
     doc.setField(FileConnector.MODIFIED, obj.lastModified());
     // s3 doesn't have object creation date
     // compression stream doesn't have size, so we can't set it here
-    if (getFileContent) {
+    if (params.shouldGetFileContent()) {
       doc.setField(FileConnector.CONTENT, is.readAllBytes());
     }
 
     return doc;
   }
 
-  private boolean isValid(S3Object obj) {
+  private boolean isValid(S3Object obj, TraversalParams params) {
     String key = obj.key();
     if (key.endsWith("/")) return false;
 
-    return shouldIncludeFile(key, includes, excludes);
+    return params.shouldIncludeFile(key);
   }
 
-  private String getFullPath(S3Object obj) {
-    return pathToStorageURI.getScheme() + "://" + bucketOrContainerName + "/" + obj.key();
+  private String getFullPath(S3Object obj, TraversalParams params) {
+    return params.getPathToStorageURI().getScheme() + "://" + getBucketOrContainerName(params) + "/" + obj.key();
   }
 
-  public static void validateOptions(Map<String, Object> cloudOptions) {
-    if (!cloudOptions.containsKey(S3_ACCESS_KEY_ID) ||
-        !cloudOptions.containsKey(S3_SECRET_ACCESS_KEY) ||
-        !cloudOptions.containsKey(S3_REGION)) {
-      throw new IllegalArgumentException("Missing '" + S3_ACCESS_KEY_ID + "' or '" + S3_SECRET_ACCESS_KEY + "' or '" + S3_REGION + "' in cloudOptions for S3StorageClient.");
-    }
+  public static boolean validOptions(Config cloudOptions) {
```

Comment:
same comment as elsewhere:
this method seems to only be called from within validateOptions()
please move this logic inside validateOptions and eliminate this static validOptions method